test/test_sparse_csr.py

```diff
@@ -2798,6 +2798,31 @@ def run_test(shape, nnz, index_type):
             run_test(shape, max(shape), index_dtype)
             run_test(shape, shape[0] * shape[1], index_dtype)
 
+    @skipMeta
+    @dtypes(*all_types_and_complex_and(torch.half, torch.bfloat16))
+    def test_sum_dim_reduce(self, device, dtype):
+        def run_test(shape, nnz, index_type):
+            sparse = self.genSparseCSRTensor(shape, nnz, dtype=dtype, device=device, index_dtype=index_dtype)
+            dense = sparse.to_dense()
+            for dim in (0, 1):
+                dense_sum = dense.sum(dim=dim)
+                sparse_sum = sparse.sum(dim=dim)
```

Comment:
reductions over sparse dimensions of sparse compressed tensors require `keepdim=True`